ResearchKit/Common/ORKPageStep.h

```diff
@@ -0,0 +1,132 @@
+
+/**
+ The `ORKPageStep` class is a concrete subclass of `ORKStep`, used for presenting a subgrouping of
+ `ORKStepViewController` views using a `UIPageViewController`.
+ 
+ To use `ORKPageStep`, instantiate the object, fill in its properties, and include it in a task. 
+ Next, create a task view controller for the task and present it.
+ 
+ The base class implementation will instatiate a read-only `ORKPageStepViewController` to display
+ a series of substeps. For each substep, the `ORKStepViewController` will be instantiated and added
+ as a child of the `UIPageViewController` contained by the parent `ORKPageStepViewController`..
+ 
+ Customization can be handled by overriding the base class implementations in either `ORKPageStep`
+ or `ORKPageStepViewController`.
+ */
+
+ORK_CLASS_AVAILABLE
+@interface ORKPageStep : ORKStep
+
+/**
+ Returns an initialized page step using the specified identifier and array of steps.
+ 
+ @param identifier  The unique identifier for the step.
+ @param steps       An array of `ORKStep` objects in the order in which they should be presented.
+ 
+ @return An initialized page step.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                             steps:(nullable NSArray<ORKStep *> *)steps NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns an initialized page step using the specified identifier and array of steps.
+ 
+ @param identifier  The unique identifier for the step.
+ @param task        The task used to run the subtask.
+ 
+ @return An initialized page step.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                          pageTask:(ORKOrderedTask *)task NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns a page step initialized from data in the given unarchiver.
+ 
+ A page step can be serialized and deserialized with `NSKeyedArchiver`. Note
+ that this serialization includes strings that might need to be localized.
+ 
+ @param aDecoder    The coder from which to initialize the ordered task.
+ 
+ @return An initialized ordered task.
+ */
+- (instancetype)initWithCoder:(NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
+
+/**
+ The subtask used to determine the next/previous steps that are in this grouping
+ */
+@property (nonatomic, copy, readonly) ORKOrderedTask *pageTask;
+
+/**
+ Returns the step after the specified step, if there is one.
+ 
+ The page view controller calls this method to determine the step to display after the specified step. 
+ The page view controller can also call this method every time the result updates, to determine if the 
+ new result changes which steps are available.
+ 
+ @param identifier      The reference step identifier. Pass `nil` to specify the first step.
+ @param result          A snapshot of the current set of results.
+ 
+ @return The step that comes after the specified step, or `nil` if there isn't one.
+ */
+- (nullable ORKStep *)stepAfterStepWithIdentifier:(nullable NSString *)identifier withResult:(ORKTaskResult *)result;
```

Comment:
@YuanZhu-apple 
This is an implementation that we are currently using in our code. The original base class implementation did not use an `ORKOrderedTask` and relied upon the methods overriding the `stepBeforeStep:result:` and `stepAfterStep:result` methods instead. Because overriding and implementing custom navigation logic was more confusing for the purposes for which @md0u80c9 wanted to use the class, I changed it to allow for initializing with an `ORKOrderedTask`.

I've gone back to using the prior implementation's public methods and instead have a subclass called `ORKPageTaskStep` that publicly exposes the pageTask property and initializer.